app/controllers/api/base_controller/manager.rb

```diff
@@ -140,7 +140,7 @@ def update_multiple_collections(is_subcollection, target, type, resources)
           update_one_collection(is_subcollection, target, type, rid, r)
         end
         raise BadRequestError, "No #{type} resources were specified for the #{action} action" if processed == 0
-        {"results" => results}
+        {"results" => results.flatten}
```

Comment:
I wonder if the flattening shouldn't be in lower level, maybe assign_tags_resource.